test/token/ERC20WithPermit.test.js

```diff
@@ -0,0 +1,1102 @@
+          const tx = await token
+            .connect(initialHolder)
+            .transfer(recipient.address, amount)
+
+          await expect(tx)
+            .to.emit(token, "Transfer")
+            .withArgs(initialHolder.address, recipient.address, amount)
+        })
+      })
+    })
+
+    context("when the recipient is the zero address", () => {
+      it("should revert", async () => {
+        await expect(
+          token.connect(initialHolder).transfer(ZERO_ADDRESS, initialSupply)
+        ).to.be.revertedWith("Transfer to the zero address")
+      })
+    })
+  })
+
+  describe("transferFrom", () => {
+    context("when the token owner is not the zero address", () => {
+      context("when the recipient is not the zero address", () => {
+        context("when the spender has enough approved balance", () => {
+          const allowance = initialSupply
+          beforeEach(async function () {
+            await token
+              .connect(initialHolder)
+              .approve(anotherAccount.address, allowance)
+          })
+
+          context("when the token owner has enough balance", () => {
+            const amount = initialSupply
+
+            it("should transfer the requested amount", async () => {
+              await token
+                .connect(anotherAccount)
+                .transferFrom(initialHolder.address, recipient.address, amount)
+
+              expect(await token.balanceOf(initialHolder.address)).to.equal(0)
+
+              expect(await token.balanceOf(recipient.address)).to.equal(amount)
+            })
+
+            it("should decrease the spender allowance", async () => {
+              await token
+                .connect(anotherAccount)
+                .transferFrom(initialHolder.address, recipient.address, amount)
+
+              expect(
+                await token.allowance(
+                  initialHolder.address,
+                  anotherAccount.address
+                )
+              ).to.equal(0)
+            })
+
+            it("should emit a transfer event", async () => {
+              const tx = await token
+                .connect(anotherAccount)
+                .transferFrom(initialHolder.address, recipient.address, amount)
+
+              await expect(tx)
+                .to.emit(token, "Transfer")
+                .withArgs(initialHolder.address, recipient.address, amount)
+            })
+
+            it("should emit an approval event", async () => {
+              const tx = await token
+                .connect(anotherAccount)
+                .transferFrom(initialHolder.address, recipient.address, amount)
+
+              await expect(tx)
+                .to.emit(token, "Approval")
+                .withArgs(
+                  initialHolder.address,
+                  anotherAccount.address,
+                  allowance.sub(amount)
+                )
+            })
+          })
+
+          context("when the token owner does not have enough balance", () => {
```

Comment:
Can we add a test case where sender sends two subsequend `transferFrom` transctions to the same `recipient`?